examples/python/src/hello-world.py

```diff
@@ -0,0 +1,212 @@
+import logging
+import os
+import json
+
+import requests
+from flask import Flask, request
+
+logging.basicConfig(level=logging.DEBUG)
+
+app = Flask(__name__, static_url_path='/static', static_folder='./static')
+
+DEFAULT_APP_PORT = 8080
+DEFAULT_APP_HOST = 'localhost'
+DEFAULT_ROOT_URL = 'http://localhost:8080'
+FORM = {
+    'title': 'I am a form!',
+    'icon': 'icon.png',
+    'fields': [
+        {
+            'type': 'text',
+            'name': 'message',
+            'label': 'message',
+            'position': 1,
+        }
+    ],
+    'submit': {
+        'path': '/submit',
+    },
+}
+
+
+@app.route('/manifest.json')
+def manifest() -> dict:
+    return {
+        'app_id': 'hello-world',
+        'display_name': 'Hello world app',
+        'homepage_url': 'https://github.com/mattermost/mattermost-plugin-apps/tree/master/examples/python/hello-world',
+        'app_type': 'http',
+        'icon': 'icon.png',
+        'requested_permissions': ['act_as_bot'],
+        'on_install': {
+            'path': '/install',
+            'expand': {
+                'app': 'all',
+            },
+        },
+        'bindings': {
+            'path': '/bindings',
+        },
+        'requested_locations': [
+            '/channel_header',
+            '/command'
+        ],
+        'root_url': os.environ.get('root_url', DEFAULT_ROOT_URL),
+    }
+
+
+@app.route('/submit', methods=['POST'])
+def on_form_submit():
+    print(request.json)
+    return {'type': 'ok', 'text': f'Hello form got submitted. Form data: {request.json["values"]}'}
+
+
+@app.route('/bindings', methods=['GET', 'POST'])
+def on_bindings() -> dict:
+    print(f'bindings called with {request.data}')
+    return {
+        'type': 'ok',
+        'data': [
+            {
+                # binding for a command
+                'location': '/command',
+                'bindings': [
+                    {
+                        'description': 'test command',
+                        'hint': '[This is testing command]',
+                        # this will be the command displayed to user as /first-command
+                        'label': 'first-command',
+                        'icon': 'icon.png',
+                        'submit': {
+                            'path': '/first_command',
+                            # expand block is optional. This is more of metadata like which channel, team this command
+                            # was called from
+                            'expand': {
+                                'app': 'all',
+                                # if you want to expand team & channel, ensure that bot is added to the team & channel
+                                # else command will fail to expand the context
+                                # 'team': 'all',
+                                # 'channel': 'all',
+                            },
```

Comment:
> if you want to expand team & channel, ensure that bot is added to the team & channel
> else command will fail to expand the context

The bot must be a member of the team and channel, otherwise the expand fails and the request is never sent to the App? It seems that we shouldn't show the command to the user in this case. cc @levb

Reply:
@levb @mickmister , 

I am able to get the `app` in the context without even adding the bot to the team/channel. And as expect expansion of `team` , `channel` failed as it is not the part of any team/channel currently. 

--- 
Steps to reproduce
1. install the app via /apps install http <url> (use the same app binding as in the current PR)
2. run the command `/first-command` and press enter to submit. The command will succeed and won't show any error. 

Sharing raw response from `ngrok` for ref, if needed.

```json
{
    "path": "/first_command",
    "expand": {
        "app": "all"
    },
    "context": {
        "app_id": "hello-world",
        "location": "/command/first-command",
        "user_agent": "webapp",
        "track_as_submit": true,
        "mattermost_site_url": "http://localhost:8066",
        "developer_mode": true,
        "app_path": "/plugins/com.mattermost.apps/apps/hello-world",
        "bot_user_id": "3xxj6snbwby19npr1tca3ys9zo",
        "bot_access_token": "3mf1popcwjnzxyeps9kqsqeuoc",
        "app": {
            "app_id": "hello-world",
            "webhook_secret": "zrp5e37iejgtixpfqtjt43ibua",
            "bot_user_id": "3xxj6snbwby19npr1tca3ys9zo",
            "bot_username": "hello-world",
            "remote_oauth2": {}
        },
        "acting_user": {
            "id": "kredsjwqwbrp9r41c86b69akno",
            "delete_at": 0,
            "username": "",
            "auth_service": "",
            "email": "",
            "nickname": "",
            "first_name": "",
            "last_name": "",
            "position": "",
            "roles": "",
            "locale": "",
            "timezone": null,
            "disable_welcome_email": false
        },
        "oauth2": {}
    },
    "raw_command": "/first-command "
}
```

Reply:
Okay so when you say "command will fail to expand the context", it's sort of a silent error, and the app still receives the request. It's just not the full request with all values. I think this is due to new changes to expand (optional vs required), depending on which version of the App framework you're working with.

Reply:
No actually "command will fail to expand the context" mean is that command will fail in the UI itself and not silent failure. And app won't receive the request. 

<img width="1589" alt="Screenshot 2022-07-25 at 10 34 52 PM" src="https://user-images.githubusercontent.com/50877070/180835376-f456fdb2-4d1d-4d11-89e8-47a578de56c5.png">